include/kokkos-utils/callbacks/EventInRegionMatcher.hpp

```diff
@@ -0,0 +1,62 @@
+#ifndef KOKKOS_UTILS_CALLBACKS_EVENTINREGIONMATCHER_HPP
+#define KOKKOS_UTILS_CALLBACKS_EVENTINREGIONMATCHER_HPP
+
+#include "kokkos-utils/callbacks/Events.hpp"
+#include "kokkos-utils/callbacks/Matcher.hpp"
+
+namespace Kokkos::utils::callbacks
+{
+
+/**
+ * @brief Matcher to select events that occur within a region.
+ *
+ * The region is selected by @ref matcher.
+ *
+ * Any event occuring between the matching @ref PushRegionEvent and its
+ * corresponding @ref PopRegionEvent will match.
+ */
+template <typename MatcherType> requires Matcher<MatcherType, Kokkos::Impl::type_list<PushRegionEvent>>
+struct EventInRegionMatcher
+{
+    static constexpr uint32_t invalid_nested_level = Kokkos::Experimental::finite_max_v<uint32_t>;
+
+    //! If the @p event matches @ref matcher, @ref matching is set to @c true.
+    bool operator()(const PushRegionEvent& event)
+    {
+        if (this->matching) {
```

Comment:
We could move the `{` to the next line to keep our way of formatting "multiline scopes". There are a few other cases in the next lines of code too.

Reply:
Indeed, and done :heavy_check_mark: